pkg/server/test/write_assertions.go

```diff
@@ -208,3 +250,134 @@ type repo
 		})
 	}
 }
+
+func TestConceptualTuplesWriteAssertionsFailure(t *testing.T, datastore storage.OpenFGADatastore) {
```

Comment:
In the previous test you added elements to the test case array. 👍 

But here you basically copied `TestWriteAssertionsFailure`. Why? You could have done what you did above, and include your new test cases in `TestWriteAssertionsFailure`

Reply:
I broke them out into separate test cases tuples vs conceptual tuples. Each are handled differently in the code (Check, WriteAssertion) and returns the same error codes and need to be tested as such and allow anyone to expand on them without impact the other.

Reply:
> tuples vs contextual tuples. 

Both are part of the WriteAssertion input. 

In general, across the codebase, we have test cases with a `request` input that tests all the various possible combinations for any API. I think this specific suite shouldn't be any different.

> Each are handled differently in the code (Check, WriteAssertion) and returns the same error codes

If we want, in the future we can come back and create a helper utility that has all the contextual tuple validation code.

Reply:
👍